src/index/base.cpp

```diff
@@ -195,10 +197,10 @@ void BaseIndex::ThreadSync()
                 LOCK(cs_main);
                 const CBlockIndex* pindex_next = NextSyncBlock(pindex, m_chainstate->m_chain);
                 if (!pindex_next) {
-                    m_best_block_index = pindex;
-                    m_synced = true;
-                    // No need to handle errors in Commit. See rationale above.
-                    Commit(GetLocator(*m_chain, pindex->GetBlockHash()));
+                    assert(pindex);
+                    notifications->blockConnected(node::MakeBlockInfo(pindex));
+                    assert(m_synced);
+                    notifications->chainStateFlushed(GetLocator(*m_chain, pindex->GetBlockHash()));
```

Comment:
e359c2666cfd205c2ea84c241bccf565670365c0: I'm a bit puzzled about what `chainStateFlushed` is for in the context of indexing. Right now it doesn't do much other than call `Commit()`. But we weren't calling it here before.

The commit title says "... to chainStateFlushed" but in reality that function is untouched and only `blockConnected` is changed.